server/postgres/src/utils.ts

```diff
@@ -71,52 +72,66 @@ export const NumericTypes = [
   core.class.Collection
 ]
 
-export async function createTables (client: postgres.Sql, domains: string[]): Promise<void> {
+export async function createTables (ctx: MeasureContext, client: postgres.Sql, domains: string[]): Promise<void> {
   const filtered = domains.filter((d) => !loadedDomains.has(d))
   if (filtered.length === 0) {
     return
   }
   const mapped = filtered.map((p) => translateDomain(p))
   const inArr = mapped.map((it) => `'${it}'`).join(', ')
-  const tables = await client.unsafe(`
+  const tables = await ctx.with('load-table', {}, () =>
+    client.unsafe(`
     SELECT table_name 
     FROM information_schema.tables 
     WHERE table_name IN (${inArr})
   `)
+  )
 
   const exists = new Set(tables.map((it) => it.table_name))
 
   await retryTxn(client, async (client) => {
+    await ctx.with('load-schemas', {}, () =>
+      getTableSchema(
+        client,
+        mapped.filter((it) => exists.has(it))
+      )
+    )
     for (const domain of mapped) {
-      if (exists.has(domain)) {
-        await getTableSchema(client, domain)
-      } else {
-        await createTable(client, domain)
+      if (!exists.has(domain)) {
+        await ctx.with('create-table', {}, () => createTable(client, domain))
       }
       loadedDomains.add(domain)
     }
   })
 }
 
-async function getTableSchema (client: postgres.Sql, domain: string): Promise<void> {
-  const res = await client.unsafe(`SELECT column_name, data_type, is_nullable
-    FROM information_schema.columns
-    WHERE table_name = '${domain}' and table_schema = 'public' ORDER BY ordinal_position ASC;
-  `)
+async function getTableSchema (client: postgres.Sql, domains: string[]): Promise<void> {
+  const res = await client.unsafe(
+    `SELECT column_name, data_type, is_nullable, table_name
+            FROM information_schema.columns
+            WHERE table_name = ANY($1) and table_schema = 'public' 
```

Comment:
[minor] Seems like can be used `in`
```suggestion
            WHERE table_name IN ($1) AND table_schema = 'public'
```

Reply:
@denis-tingaikin any difference between?

Reply:
No. `in` for cases when we look for a match in a list can be more readable for devs. So comment minor, can be ignored.

Reply:
We should use ANY in this concrete context.